src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStore.cs

```diff
@@ -27,19 +31,23 @@ internal class SqlIndexDataStore : IIndexDataStore
     {
         private readonly SqlIndexSchema _sqlServerIndexSchema;
         private readonly SqlConnectionWrapperFactory _sqlConnectionFactoryWrapper;
+        private readonly DicomTagParser _dicomTagParser;
 
         public SqlIndexDataStore(
             SqlIndexSchema indexSchema,
-            SqlConnectionWrapperFactory sqlConnectionWrapperFactory)
+            SqlConnectionWrapperFactory sqlConnectionWrapperFactory,
+            DicomTagParser dicomTagParser)
```

Comment:
can you avoid this here since this is not SQL pacific? Store logic needs to be only specific to the store, rest should be calculated at the service level.